plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,273 @@
+    >>> c_s = pfp.ion_sound_speed(T_e=T_e, T_i=T_i, n_e=z * n, ion='p+')
+    >>> v_A = pfp.Alfven_speed( B, n, ion='p+')
+    >>> omega_ci = pfp.gyrofrequency(B=B, particle='p+', signed=False, Z=z)
+    >>> omega = tfds(n=n, B=B, T_e=T_e, T_i=T_i, theta=theta, z=z, k=k_arr)
+    >>> omega['fast_mode'][:,40]
+    <Quantity [1.61176312e-02, 1.77335334e-01, 3.38688590e-01, ...,
+               1.52030361e+03, 1.52045553e+03, 1.52060745e+03] rad / s>
+    """
+
+    # Required derived parameters
+    # Compute the ion sound speed using the function from
+    # plasmapy.formulary.parameters
+    c_s = pfp.ion_sound_speed(
+        T_e=T_e, T_i=T_i, n_e=z * n, gamma_e=gamma_e, gamma_i=gamma_i, ion=ion
+    )
+
+    # Compute the ion Alfven speed using the function from
+    # plasmapy.formulary.parameters
+    v_A = pfp.Alfven_speed(B, n, ion=ion)
+
+    # Compute the ion gyro frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_ci = pfp.gyrofrequency(B=B, particle="p+", signed=False, Z=z)
+
+    # Compute the electron plasma frequency using the function from
+    # plasmapy.formulary.parameters
+    omega_pe = pfp.plasma_frequency(n=n, particle="e-", z_mean=z)
+
+    # Compute the dimensionless parameters corresponding to equation 32 of
+    # Bellan2012JGR
+    alpha = (np.cos(theta.to("rad")) ** 2).value
+    beta = (c_s ** 2 / v_A ** 2).value
+
+    # Create a meshgrid of direction of propagation and the wavenumber
+    alphav, kv = np.meshgrid(alpha, k)
+
+    Lambda = (kv ** 2 * v_A ** 2 / omega_ci ** 2).value
+
+    # Compute the dimensionless parameters corresponding to equation 2 of
+    # Bellan2012JGR
+    Q = 1 + (kv ** 2 * c ** 2 / omega_pe ** 2).value
+
+    # Compute the dimensionless parameters corresponding to equation 35 of
+    # Bellan2012JGR
+    A = (Q + Q ** 2 * beta + Q * alphav + alphav * Lambda) / Q ** 2
+    B = alphav * (1 + 2 * Q * beta + Lambda * beta) / Q ** 2
+    C = alphav ** 2 * beta / Q ** 2
+
+    # Compute the dimensionless parameters corresponding to equation 36 of
+    # Bellan2012JGR
+    p = (3 * B - A ** 2) / 3
+    q = (9 * A * B - 2 * A ** 3 - 27 * C) / 27
+
+    # These correspond to different parts of equation 38 of
+    # Bellan2012JGR
+    R = 2 * Lambda * np.lib.scimath.sqrt(-p / 3)
+    S = 3 * q / (2 * p) * np.lib.scimath.sqrt(-3 / p)
+    T = Lambda * A / 3
+
+    # List out the three wave modes for which this function gives the
+    # frequencies
+    keys = ["fast_mode", "alfven_mode", "acoustic_mode"]
+
+    # Create a dictionary with the wave mode names as its keys
+    omega = dict.fromkeys(keys)
+
+    # Compute the value of  omega for each key and for each value of wavenumber
+    # and direction of propagation
+    for (ind, key) in zip(range(3), keys):
+        # The solution corresponding to equation 38
+        omega[key] = omega_ci * np.lib.scimath.sqrt(
+            R * np.cos(1 / 3 * np.lib.scimath.arccos(S) - 2 * np.pi / 3 * ind) + T
+        )
```

Comment:
```suggestion
    # Bellan2012JGR params equation 38
    R = 2 * Lambda * np.emath.sqrt(-p / 3)
    S = 3 * q / (2 * p) * np.emath.sqrt(-3 / p)
    T = Lambda * A / 3
    omega = {}
    for ind, key in enumerate(("fast_mode", "alfven_mode", "acoustic_mode")):
        # The solution corresponding to equation 38
        w = omega_ci * np.emath.sqrt(
            R * np.cos(1 / 3 * np.emath.arccos(S) - 2 * np.pi / 3 * ind) + T
        )
        omega[key] = w.squeeze()

        # check for violation of dispersion relation assumptions
        # (i.e. low-frequency, w/kc << 0.1)
        wkc_max = np.max(w.value / (kv * c.value))
        if wkc_max > 0.1:
            warn(
                f"The {key} calculation produced a high-frequency wave (w/kc == "
                f"{wkc_max}), which violates the low-frequency (w/kc << 1) assumption "
                f"of the dispersion relation.",
                PhysicsWarning,
            )
```

Again, this does not replace the original functionality of the code but does...

1. replace use of `numpy.lib.scimath.*` with `numpy.emath.*` as per [`numpy`'s suggestion](https://numpy.org/devdocs/reference/routines.emath.html?highlight=emath#module-numpy.lib.scimath)
2. streamlines the definition of the dictionary to improve readability...this part is really a personal so choose the method you prefer
3. add a warning for when/if violates w/kc << 1 (a key assumption behind the dispersion relation)